### DISCUSSION TITLE
AsDiscreted extremely slow

### DISCUSSION THREAD

**Discussion by sunshichen** [Project-MONAI/MONAI#3859] [Q&A]
I tried to test monailabel follow the official guidline on a remote server and use 3Dslicer as annotation client. How ever it seems unusual that AsDiscreted in post-processing is extremely slow, the log as below:

![1645766526(1)](https://user-images.githubusercontent.com/23454839/155658705-90d0f661-0754-47e0-bf7c-739c10e2435b.png)

I used deepedit_multilabel sample on Ubuntu18.04 with Nvidia GTX 1080Ti.

**Comment by Nic-Ma** [comment]
Hi @sunshichen ,

Could you please help provide the full test program? Then @SachidanandAlle maybe can help figure out more ideas.

Thanks in advance.

**Comment by rijobro** [comment]
Duplicate of https://github.com/Project-MONAI/MONAILabel/issues/604.

**Comment by SachidanandAlle** [comment]
The quick fix for this shall be.. use `EnsureTyped(keys="pred", device="cuda")` as the first post transform

You can refer: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/deepedit_multilabel/lib/infer.py#L97

This can help to speed up the post transform using GPU..
